src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -461,10 +473,11 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
             return result;
         }
 
-        /// <summary>Resyncs the peer whenever state is changed.</summary>
+        /// <summary>Sync when handshake is finished.</summary>
         private async Task OnStateChangedAsync(INetworkPeer peer, NetworkPeerState oldState)
         {
-            await this.ResyncAsync().ConfigureAwait(false);
+            if (peer.State == NetworkPeerState.HandShaked)
```

Comment:
State is already checked inside ResyncAsync so not a big deal here

Reply:
Yeah, added another check because this always bothers me. It's weird to resync on every state change.